File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.java
class Solution {
    public String solution(String my_string, int num1, int num2) {
        String answer = "";
        StringBuilder b = new StringBuilder(my_string);
        b.setCharAt(num1, my_string.charAt(num2));
        b.setCharAt(num2, my_string.charAt(num1));
        
        answer = b.toString();
        
        return answer;
    }
}

File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* my_string, int num1, int num2) {
    char* answer = (char*)malloc(strlen(my_string) * sizeof(char));
    strcpy(answer, my_string);
    
    answer[num1] = my_string[num2];
    answer[num2] = my_string[num1];
    
    return answer;
}